src/lib/y2network/sysconfig/type_detector.rb

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface_type"
+require "y2network/type_detector"
+
+Yast.import "NetworkInterfaces"
+
+module Y2Network
+  module Sysconfig
+    # Detects type of given interface. New implementation of what was in
+    # @see Yast::NetworkInterfaces.GetType
+    class TypeDetector < Y2Network::TypeDetector
+      class << self
+      private
+
+        KEY_TO_TYPE = {
+          "ETHERDEVICE"   => InterfaceType::VLAN,
+          "WIRELESS_MODE" => InterfaceType::WIRELESS,
+          "MODEM_DEVICE"  => InterfaceType::PPP
+        }.freeze
+
+        # Checks wheter iface type can be recognized by interface configuration
+        def type_by_config(iface)
+          devmap = devmap(iface)
+          return nil if !devmap
+
+          type_by_key_value(devmap) ||
+            type_by_key_existence(devmap) ||
+            type_from_interfacetype(devmap) ||
+            type_by_name(iface)
+        end
+
+        # Provides interface's sysconfig configuration
+        #
+        # @return [Hash{String => Object}] parsed configuration
+        def devmap(iface)
+          scr_path = ".network.value.\"#{iface}\""
+          values = Yast::SCR.Dir(Yast::Path.new(scr_path))
+
+          # provide configuration in canonicalized format
+          devmap = Yast::NetworkInterfaces.generate_config(scr_path, values)
```

Comment:
cannot we use what we currently have in connection config? using for generating config network interfaces looks strange for me.

Reply:
ok I'll try to polish it a bit.